lib/src/components/NxDropdown/__tests__/NxDropdown.test.tsx

```diff
@@ -5,357 +5,301 @@
+  it('sets the provided className', function() {
+    const el = renderEl()!;
+    const customEl = renderEl({ className: 'foo' })!;
+
+    expect(customEl).toHaveClass('foo');
+
+    for (const cls of Array.from(el.classList)) {
+      expect(customEl).toHaveClass(cls);
+    }
   });
 
-  it('disables the button (and the toggle fn) when the disabled prop is supplied', function() {
-    const toggleFn = jest.fn(),
-        component = getMountedComponent({ onToggleCollapse: toggleFn, disabled: true }),
-        button = component.find(NxButton);
+  it('disables the button (and the toggle fn) when the disabled prop is supplied', async function() {
+    const user = userEvent.setup();
+    const toggleFn = jest.fn();
+    renderEl({ onToggleCollapse: toggleFn, disabled: true });
+    const button = screen.getByRole('button');
+    expect(button).toHaveClass('disabled');
+
+    await user.click(button);
 
-    expect(button).toHaveClassName('disabled');
-    button.simulate('click');
     expect(toggleFn).not.toHaveBeenCalled();
   });
 
-  it('uses VDOM as label if supplied', function() {
-    const label = <NxFontAwesomeIcon icon={faTrash} />,
-        component = getMountedComponent({ label });
-    const button = component.find(NxButton);
+  it('renders react element as label if supplied', function() {
+    const label = <NxFontAwesomeIcon icon={faTrash} />;
+    const { container } = quickRender({ label });
 
-    expect(button.childAt(0)).toContainReact(label);
+    expect(container.querySelector('.nx-icon.fa-trash')).toBeInTheDocument();
   });
 
-  it('passes extra props', function() {
-    const component = getShallowComponent({ id: 'some-id', title: 'title-prop' });
-    expect(component).toHaveProp('id', 'some-id');
-    expect(component).toHaveProp('title', 'title-prop');
+  it('renders provided attributes', function() {
+    const component = renderEl({ id: 'some-id', title: 'title-prop' });
+    expect(component).toHaveAttribute('id', 'some-id');
+    expect(component).toHaveAttribute('title', 'title-prop');
   });
 
-  it('renders the children within the NxDropdownMenu, wrapping nx-dropdown-links and nx-dropdown-buttons ' +
-      'in NxOveflowTooltip', function() {
+  it('renders the children within the NxDropdownMenu in the specified order', async function() {
     const children = [
-      <a id="link1" key="1">Link1</a>,
-      <a id="link2" className="nx-dropdown-link" key="2">Link2</a>,
-      <button id="link3" className="nx-dropdown-button" key="3">Link3</button>,
-      <button id="link4" className="nx-dropdown-right-button" key="4">Link4</button>
+      <a data-testid="menu-child" id="link1" key="1">Link1</a>,
+      <a data-testid="menu-child" id="link2" className="nx-dropdown-link" key="2">Link2</a>,
+      <button data-testid="menu-child" id="link3" className="nx-dropdown-button" key="3">Link3</button>,
+      <button data-testid="menu-child" id="link4" className="nx-dropdown-right-button" key="4">Link4</button>
```

Comment:
The ids and classes aren't really relevant to this test; they can be removed

Reply:
> The ids and classes aren't really relevant to this test; they can be removed

Cleaned up here: b02bf2ba530f04db03a59c4b6c75c64294cbb19d